src/mlpack/tests/main_tests/load_save_image_test.cpp

```diff
@@ -0,0 +1,213 @@
+  SetInputParam("channel", 3);
+
+  mlpackMain();
+  arma::mat randomOutput = CLI::GetParam<arma::mat>("output");
+
+  SetInputParam<vector<string>>("input", {"test_image.png", "test_image.png"});
+  SetInputParam("input_model",
+                CLI::GetParam<ImageInfo*>("output_model"));
+
+  mlpackMain();
+  arma::mat savedOutput = CLI::GetParam<arma::mat>("output");
+  CheckMatrices(randomOutput, savedOutput);
+}
+
+/**
+ * Check transpose option give two different output.
+ */
+BOOST_AUTO_TEST_CASE(TransposeTest)
+{
+  SetInputParam<vector<string>>("input", {"test_image.png", "test_image.png"});
+  SetInputParam("height", 50);
+  SetInputParam("width", 50);
+  SetInputParam("channel", 3);
+
+  mlpackMain();
+  arma::mat normalOutput = CLI::GetParam<arma::mat>("output");
+
+  SetInputParam<vector<string>>("input", {"test_image.png", "test_image.png"});
+  SetInputParam("input_model",
+                CLI::GetParam<ImageInfo*>("output_model"));
+  SetInputParam("transpose", true);
+  mlpackMain();
+  arma::mat transposeOutput = CLI::GetParam<arma::mat>("output");
+
+  CheckMatricesNotEqual(normalOutput, transposeOutput);
+  BOOST_REQUIRE_EQUAL(normalOutput.n_rows, transposeOutput.n_rows);
+  BOOST_REQUIRE_EQUAL(normalOutput.n_cols, transposeOutput.n_cols);
+}
+
+/**
+ * Check whether binding throws error if height, width or channel are not
+ * specified.
+ */
+BOOST_AUTO_TEST_CASE(IncompleteTest)
+{
+  SetInputParam<vector<string>>("input", {"test_image.png", "test_image.png"});
+  SetInputParam("height", 50);
+  SetInputParam("width", 50);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Check for invalid height values.
+ */
+BOOST_AUTO_TEST_CASE(InvalidInputTest)
+{
+  SetInputParam<vector<string>>("input", {"test_image.png", "test_image.png"});
+  SetInputParam("height", -50);
+  SetInputParam("width", 50);
+  SetInputParam("channel", 3);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Check for invalid width values.
+ */
+BOOST_AUTO_TEST_CASE(InvalidWidthTest)
+{
+  SetInputParam<vector<string>>("input", {"test_image.png", "test_image.png"});
+  SetInputParam("height", 50);
+  SetInputParam("width", -50);
+  SetInputParam("channel", 3);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Check for invalid channel values.
+ */
+BOOST_AUTO_TEST_CASE(InvalidChannelTest)
+{
+  SetInputParam<vector<string>>("input", {"test_image.png", "test_image.png"});
+  SetInputParam("height", 50);
+  SetInputParam("width", 50);
+  SetInputParam("channel", -1);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Check for invalid input values.
+ */
+BOOST_AUTO_TEST_CASE(EmptyinputTest)
```

Comment:
```suggestion
BOOST_AUTO_TEST_CASE(EmptyInputTest)
```
Just a little bit of pedantry. :)

Reply:
Done